schema/draft/schema.ts

```diff
@@ -1462,19 +1463,57 @@ export interface StreamPollRequest extends Request {
 }
 
 /**
- * A response to the `stream/poll` request.
+ * The status of a stream.
  */
-export interface StreamPollResponse extends Response {
+export interface StreamStatus {
   /**
-   * Messages and notifications to be delivered to the client on the stream.
-   * The client should respond to any pending requests found in this list.
+   * The ID of the stream.
    */
-  messages: (ServerNotification | ServerRequest)[];
+  streamId: string;
 
   /**
-   * Optionally-updated properties for the stream.
+   * The current status of the stream.
    */
-  stream?: String;
+  status: "live" | "completed" | "abandoned";
+
+  /**
+   * Whether the stream has pending messages.
+   */
+  pendingMessages: boolean;
+
+  /**
+   * Whether the stream's pending messages include a server-sent request.
+   */
+  hasRequest: boolean;
+
+  /**
+   * Whether the stream's pending messages include an error result.
+   */
+  hasError: boolean;
+}
+
+/**
+ * A response to a `stream/poll` request.
+ */
+export interface StreamPollResult extends Result, StreamStatus {
+}
+
+/**
+ * A request from the client to the server to check the status of all current
+ * streams.
+ *
+ * The server SHOULD reset each stream's abandonment timer when responding to
+ * this request.
+ */
+export interface StreamPollAllRequest extends PaginatedRequest {
+  method: "stream/poll/all";
+}
+
+/**
+ * A response to a `stream/poll/all` request.
+ */
+export interface StreamPollAllResult extends PaginatedResult {
+  statuses: StreamStatus[];
```

Comment:
We may also want a way to cancel stream subscriptions. I can imagine a popular flow for client is to poll this occasionally, have short-lived subscriptions to each stream, which are cancelled after a period of inactivity and may be re-subscribed based on polled status.

Reply:
My mental model is that a server would immediately return an (empty) JSON-RPC response for a `StreamResumeRequest`, but continue to stream messages on that connection (e.g. via SSE).  If the client wants to "cancel", so to speak, it would just close the connection.

Do you have a different model in mind?

Reply:
I think this is actually fine as it exists for now... Later if/when we have stream 'types' (as I propose in https://github.com/connor4312/mcp-spec/pull/1 for A2A streams, could also have a tool response stream type) I think we will want some way to cancel those without closing the entire connection.